AwsEncryptionSDK/runtimes/net/Examples/Keyring/AwsKmsHierarchical/AwsKmsHierarchicalKeyring.cs

```diff
@@ -34,175 +37,170 @@
+        //    To create a KeyStore table, you need:
+        //     - "dynamodb:DescribeTable",
+        //     - "dynamodb:CreateTable",
+       
+        // This will NOT create a Key Store but check that the configured table name exists and it validates
+        // its construction. In order to check the construction is correct the configured IAM role
+        // MUST have `DescribeTable` permission on the Key Store resource.
+        keyStore.CreateKeyStore(new CreateKeyStoreInput());
+        
+        // We have already created a Table with the specified configuration.
+        // For testing purposes we will not create a table when we run this example.
         
-        // Create a branch key supplier that maps the branch key id to a more readable format
-        var branchKeySupplier = new ExampleBranchKeySupplier(branchKeyIdA, branchKeyIdB);
+        // 3. Create a new branch key and beacon key in our KeyStore.
+        //    Both the branch key and the beacon key will share an Id.
+        //    This creation is eventually consistent. See the CreateBranchKey
+        //    Example for how to populate this table.
+        //    To create a Branch Key and a Beacon Key you need:
+        //    - "dynamodb:PutItem",
+        //    - "dynamodb:ConditionCheckItem",
+        //    - "dynamodb:UpdateItem" 
         
-        // Create an AWS Hierarchical Keyring with the branch key id supplier
+        // var branchKeyId = CreateBranchKey.createBranchKey();
+        
+        // For testing purposes we will not create another Branch Key when we run this example.
+        // We will use an existing branch key created using the above code to run this
+        // example.
+        
+        // THIS IS A PUBLIC RESOURCE DO NOT USE IN A PRODUCTION ENVIRONMENT
+        var branchKeyId = "43574aa0-de30-424e-bad4-0b06f6e89478";
+        
+        // 4. Create the Hierarchical Keyring, with a static Branch Key ID.
+        //    With this configuration, the AWS SDK Client ultimately configured will be capable
+        //    of encrypting or decrypting items for either tenant (assuming correct KMS access).
+        //    We are restricting the keyring to only branch key by statically configuring
+        //    it with a branch key id. For an examples on how to decide on which branch key to 
+        //    use see the `AwsKmsHierarchicalKeyringBranhcKeySupplier` example in this directory.
         var createKeyringInput = new CreateAwsKmsHierarchicalKeyringInput
         {
-            KeyStore = keystore,
-            // This branchKeyId you have configured your keyring with MUST be decrypted by the 
-            // KMS config in the keystore and therefore MUST have the right permissions.
-            BranchKeyIdSupplier = branchKeySupplier,
+            KeyStore = keyStore,
+            BranchKeyId = branchKeyId,
             // The value provided to `EntryCapacity` dictates how many branch keys will be held locally
```

Comment:
Issue: Is this true?
I think that multiple of versions of the same Branch Key
count towards the Entry Capacity.

```suggestion
            // The value provided to `EntryCapacity` dictates how many branch key versions will be held locally
```